devito/data/data.py

```diff
@@ -132,16 +141,82 @@ def _global(self, glb_idx, decomposition):
         ret._is_distributed = any(i is not None for i in decomposition)
         return ret
 
+    def check_slicing(func):
+        """Check if __getitem__/__setitem__ may require communication across MPI ranks."""
```

Comment:
does this PR support the case
```
a[0] = a[20]
```
ie, doing the right thing regardless of whether `0` and `20` belong to the same of different processes?

if yes, then maybe use another name instead of ``check_slicing`` ?

Reply:
also, it should be a private method

Reply:
Yes, `a[0] = a[20]` is supported.  Hmm, could change to `check_idx`?

Reply:
yep, now it's OK